src/Controls/StreetControls.js

```diff
@@ -102,19 +204,114 @@ class StreetControls extends FirstPersonControls {
+
+        // can't click on a wall that is at 1km distance.
+        if (distance < this.wallMaxDistance) {
+            this.animateCameraLookAt(position, this.animationDurationWall);
+        }
+    }
+
+    /**
+     * Animate the camera to make it look at a position, in a given time
+     *
+     * @param { THREE.Vector3 }  position - Position to look at
+     * @param { number } time - Time in millisecond
+     */
+    animateCameraLookAt(position, time) {
+        // stop existing animation
+        this.stopTweensAnimations();
+        // prepare start point and end point
+        this.start = this.camera.clone();
+        this.end = this.camera.clone();
+        this.end.lookAt(position);
+        this.tween = new TWEEN.Tween({ t: 0 }, this.tweenGroup).to({ t: 1 }, time)
+            .easing(TWEEN.Easing.Quadratic.Out)
+            .onComplete(this.onComplete.bind(this))
+            .onUpdate((d) => {
+                // 'manually' slerp the Quaternion to avoid rotation issues
+                THREE.Quaternion.slerp(this.start.quaternion, this.end.quaternion, this.camera.quaternion, d.t);
+            })
+            .start();
+
+        this.animationFrameRequester = () => {
+            this.tweenGroup.update();
+            // call reset from super class FirsPersonControls to make mouse rotation managed by FirstPersonControl still aligned
+            this.reset();
+            this.view.notifyChange(this.camera);
+        };
+
+        this.view.addFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, this.animationFrameRequester);
+    }
+
+    /**
+     * Move the camera smoothly to the position, in a given time.
      *
-     * @param { THREE.Vector3 }  positionTo  Destination of the movement.
+     * @param { THREE.Vector3 }  position - Destination of the movement.
+     * @param { number } time - Time in millisecond
      */
-    moveCameraTo(positionTo) {
-        if (!positionTo) {
+    moveCameraTo(position, time) {
+        if (!position) {
             return;
         }
-        this.positionFrom.copy(this.camera.position);
-        this.positionTo = positionTo;
-        this.player.play(this.animationMoveCamera);
+
+        this.stopTweensAnimations();
+
+        this.tween = new TWEEN.Tween(this.camera.position, this.tweenGroup) // Create a new tween that modifies camera position
+            .to(position, time)
+            .easing(TWEEN.Easing.Quadratic.Out) // Use an easing function to make the animation smooth.
+            .onComplete(this.onComplete.bind(this))
+            .start();
+
+        this.animationFrameRequester = () => {
+            this.tweenGroup.update();
+
+            this.view.notifyChange(this.camera);
+        };
+
+        this.view.addFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, this.animationFrameRequester);
+    }
+
+    onComplete() {
+        this.view.removeFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, this.animationFrameRequester);
+    }
+
+    stopTweensAnimations() {
```

Comment:
could you rename in `stopAnimation`? the user doesn't know it's `tween`.
could add `this.view.removeFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, this.animationFrameRequester);` in `stopTweensAnimations`? 

or merge `onComplete` and `stopTweensAnimations`...

Reply:
yes, I've renamed to stopAnimation. And I 've remove onComplete because it's not need to be exposed.

Reply:
`this.view.removeFrameRequester` isn't removed in all case

Reply:
yes, thanks, updated